lib/date_time/date_time.c

```diff
@@ -126,18 +75,23 @@ static int current_time_check(void)
 	return 0;
 }
 
-static void date_time_store(int64_t curr_time_ms)
+void date_time_store(int64_t curr_time_ms, enum date_time_evt_type time_source)
 {
 	struct timespec tp = { 0 };
 	struct tm ltm = { 0 };
 	int ret;
 
+	date_time_last_update_uptime = k_uptime_get();
+
+	date_time_schedule_update();
+
 	tp.tv_sec = curr_time_ms / 1000;
 	tp.tv_nsec = (curr_time_ms % 1000) * 1000000;
 
 	ret = clock_settime(CLOCK_REALTIME, &tp);
 	if (ret != 0) {
 		LOG_ERR("Could not set system time, %d", ret);
+		date_time_notify_event(DATE_TIME_NOT_OBTAINED);
```

Comment:
Added this one because we don't store the time when `clock_settime()` fails. Also calling `date_time_schedule_update()` before this if-statement so that updates don't stop.